test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -270,15 +270,17 @@ u256 EVMInstructionInterpreter::eval(
 		return m_state.gaslimit;
 	// --------------- memory / storage / logs ---------------
 	case Instruction::MLOAD:
-		accessMemory(arg[0], 0x20);
-		return readMemoryWord(arg[0]);
+		if (accessMemory(arg[0], 0x20))
+			return readMemoryWord(arg[0]);
+		else
+			return 0;
 	case Instruction::MSTORE:
-		accessMemory(arg[0], 0x20);
-		writeMemoryWord(arg[0], arg[1]);
+		if (accessMemory(arg[0], 0x20))
+			writeMemoryWord(arg[0], arg[1]);
 		return 0;
 	case Instruction::MSTORE8:
-		accessMemory(arg[0], 1);
-		m_state.memory[arg[0]] = uint8_t(arg[1] & 0xff);
+		if (accessMemory(arg[0], 1))
+			m_state.memory[arg[0]] = uint8_t(arg[1] & 0xff);
```

Comment:
There are still quite a few unguarded `accessMemory()` calls in this file. I guess we're assuming that we're nowhere close to the edge there? Should we replace them with something like `soltestAssert(accessMemory())` to make sure we're not missing any other cases?

Reply:
> I guess we're assuming that we're nowhere close to the edge there?

Each of those unguarded `accessMemory()` calls is in conjunction with a `logTrace()` call that writes to the interpreter trace. For this reason, we can be sure that trace of such calls is not silently ignored.

A `soltestAssert(accessMemory())` would not be useful since it's failure (not accessing bounded memory) does not indicate a bug, rather could lead to lots of false positives.